apps/web/components/booking/bookingActions.ts

```diff
@@ -163,6 +163,27 @@ export function getEditEventActions(context: BookingActionContext): ActionType[]
   return actions.filter(Boolean) as ActionType[];
 }
 
+export function getReportAction(context: BookingActionContext): ActionType | null {
+  const { booking, isCancelled, isRejected, t } = context;
+
+  if (booking.reportedByCurrentUser) {
+    return null;
+  }
+
+  // Don't allow reporting cancelled or rejected bookings
+  if (isCancelled || isRejected) {
+    return null;
+  }
```

Comment:
We should still allow this in case the user decides to reject prior to flagging as spam